tests/unit/fem/test_get_value.cpp

```diff
@@ -1667,19 +1667,19 @@ TEST_CASE("2D GetVectorValue",
                   dgv_x.GetVectorValue(e, ip, dgv_gvv_val);
                   dgi_x.GetVectorValue(e, ip, dgi_gvv_val);
 
-                  double  h1_gfc_dist = Distance(f_val,  h1_gfc_val, dim);
-                  double  nd_gfc_dist = Distance(f_val,  nd_gfc_val, dim);
-                  double  rt_gfc_dist = Distance(f_val,  rt_gfc_val, dim);
-                  double  l2_gfc_dist = Distance(f_val,  l2_gfc_val, dim);
-                  double dgv_gfc_dist = Distance(f_val, dgv_gfc_val, dim);
-                  double dgi_gfc_dist = Distance(f_val, dgi_gfc_val, dim);
-
-                  double  h1_gvv_dist = Distance(f_val,  h1_gvv_val, dim);
-                  double  nd_gvv_dist = Distance(f_val,  nd_gvv_val, dim);
-                  double  rt_gvv_dist = Distance(f_val,  rt_gvv_val, dim);
-                  double  l2_gvv_dist = Distance(f_val,  l2_gvv_val, dim);
-                  double dgv_gvv_dist = Distance(f_val, dgv_gvv_val, dim);
-                  double dgi_gvv_dist = Distance(f_val, dgi_gvv_val, dim);
+                  double  h1_gfc_dist = Distance(f_val.HostRead(),  h1_gfc_val.HostRead(), dim);
```

Comment:
This is a bit cumbersome. @v-dobrev, @tzanio, and @pazner  could we overload `mfem::Distance` (and related functions) with a pair of `mfem::Vector` arguments and move the calls to `HostRead` inside these overloads?

Reply:
We can definitely do that, see PR #3327. If that looks good to the reviewers I can merge it in here.